narwhals/dataframe.py

```diff
@@ -186,17 +186,31 @@ def join(
         how: Literal["inner", "left", "cross", "semi", "anti"] = "inner",
         left_on: str | list[str] | None = None,
         right_on: str | list[str] | None = None,
+        on: str | list[str] | None = None,
     ) -> Self:
         _supported_joins = ("inner", "left", "cross", "anti", "semi")
 
         if how not in _supported_joins:
             msg = f"Only the following join strategies are supported: {_supported_joins}; found '{how}'."
             raise NotImplementedError(msg)
 
-        if how == "cross" and (left_on or right_on):
-            msg = "Can not pass left_on, right_on for cross join"
+        if how == "cross" and (left_on or right_on or on):
+            msg = "Can not pass `left_on`, `right_on` or `on` keys for cross join"
             raise ValueError(msg)
 
+        if (how != "cross" and on is None) and (left_on is None or right_on is None):
```

Comment:
I can't really make sense of the parentheses like this, maybe it's just me, but how about
```
if how != "cross" and (on is None and (left_on is None or right_on is None)):
```